### PR TITLE
Color chromatic lerp function

### PR DESCRIPTION
**Pull request by Remi-Tribia**
Adds a chromatic lerp (maybe could be simply renamed lerpHSL) which interpolates in HSL and not straight from one color to the other in RGB like the original lerp. Other than that it works exactly like lerp()

Imgur

### CODE REVIEW

**Comment by WestLangley** [comment]
I definitely support this, including the name change.

There is considerable literature on this topic, and there may be a better implementation, but this is a good start.

Please rewrite your code to prevent console warnings:
```javascript
lerpHSL: function () {

	var hslA = { h: 0, s: 0, l: 0 };
	var hslB = { h: 0, s: 0, l: 0 };

	return function lerpHSL( color, alpha ) {

		this.getHSL( hslA );
		color.getHSL( hslB );

		var h = _Math.lerp( hslA.h, hslB.h, alpha );
		var s = _Math.lerp( hslA.s, hslB.s, alpha );
		var l = _Math.lerp( hslA.l, hslB.l, alpha );

		this.setHSL( h, s, l );

		return this;

	};

}(),
```

**Comment by WestLangley** [comment]
For a separate PR, `alpha` is probably a poor choice for a variable name in this case. `slerp()` uses `t`.

**Comment by Remi-Tribia** [comment]
I'm not used to PR, I should commit your changes and leave `alpha` for now or change it to `t` ? I used alpha so it was consistant with the lerp just above, but I agree this is not ideal.

**Comment by WestLangley** [comment]
Don't worry about `alpha` for now. You can just commit my suggested changes.

File: src/math/Color.js [hunk 1]
```diff
@@ -541,6 +541,28 @@ Object.assign( Color.prototype, {
...
+		var hslB = { h: 0, s: 0, l: 0 };
+
+		return function lerpHSL( color, alpha ) {
+
+			this.getHSL( hslA );
+			color.getHSL( hslB );
+
+			var h = _Math.lerp( hslA.h, hslB.h, alpha );
+			var s = _Math.lerp( hslA.s, hslB.s, alpha );
+			var l = _Math.lerp( hslA.l, hslB.l, alpha );
```
Comment (WestLangley) [inline src/math/Color.js]: @Mugen87 `Math` or `_Math`? Both are used in this file.

Comment (Mugen87) [reply src/math/Color.js]: `_Math`. [Math](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math) is the built-in object in JavaScript which has no `lerp()` function.

**Comment by Mugen87** [comment]
Please add the new method to the existing doc page: https://threejs.org/docs/index.html#api/math/Color

**Comment by Remi-Tribia** [comment]
Done.

Now I'm thinking it would be nice to have a flag to specify in which direction we interpolate the hues or maybe to specify if we want to interpolate using the shortest path.

*PS: As I said above, I'm not used to pull requests, I see the doc page editing is added automatically here, are all the commits I make on my fork being added here? I need to close to prevent it?*

**Comment by haroldiedema** [comment]
> PS: As I said above, I'm not used to pull requests, I see the doc page editing is added automatically here, are all the commits I make on my fork are being added here? Do I need to close the PR to prevent it?

Any commit you `push` to your fork will be added here automatically. So far so good.

**Comment by WestLangley** [comment]
>Now I'm thinking it would be nice to have a flag to specify in which direction we interpolate the hues or maybe to specify if we want to interpolate using the shortest path.

Probably, always use the shortest path. That is why I said "this is a good start".

**Comment by Remi-Tribia** [comment]
@WestLangley Well in the example picture in my first post, this is not the shortest path, but that was actually what I was looking for.

@haroldiedema Thanks! What about the eventuality of commiting things related to another PR? Sorry about the off topic...

**Comment by haroldiedema** [comment]
> Thanks! What about the eventuality of committing things related to another PR? Sorry about the off topic...

Take your time to get familiar with the git forking workflow. 
A quick google search does wonders: https://gist.github.com/Chaser324/ce0505fbed06b947d962

Next time you make a PR, make sure you create a new _branch_ before tinkering. Once done, you push the branch and simply switch back to master (or dev in this case) and create a new branch for another PR.

(no more offtopic discussions here though, there are plenty 'tutorials' about this workflow that you can find by googling "git pull request workflow" or "git fork workflow", etc.)

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
Here is a version of `lerpHSL()` which is optionally cyclic in hue. I am not particularly happy with the behavior, but here it is for future reference.

```javascript
lerpHSL: function () {

	var hslA = { h: 0, s: 0, l: 0 };
	var hslB = { h: 0, s: 0, l: 0 };

	return function lerpHSL( color, alpha, cyclic ) {

		this.getHSL( hslA );
		color.getHSL( hslB );

		var hA = hslA.h;
		var hB = hslB.h;

		if ( cyclic && Math.abs( hB - hA ) > 0.5 ) {

			hB += ( hA > hB ) ? 1 : - 1;

		}

		var h = _Math.lerp( hA, hB, alpha );
		var s = _Math.lerp( hslA.s, hslB.s, alpha );
		var l = _Math.lerp( hslA.l, hslB.l, alpha );

		if ( h < 0 ) h += 1;
		if ( h > 1 ) h -= 1;

		this.setHSL( h, s, l );

		return this;

	};

}(),
```

**Comment by mrdoob** [comment]
@WestLangley What is it that you're not happy with?

**Comment by WestLangley** [comment]
@mrdoob Well, I thought it was hard to get nice palettes. But maybe I was expecting too much -- or maybe I didn't try hard enough. :-)

I can add this feature if you like.

**Comment by mrdoob** [comment]
Maybe we can wait until someone asks for it 😇

**Pull request merged**